patches/tModLoader/Terraria/ModLoader/Core/ModOrganizer.cs

```diff
@@ -138,7 +138,12 @@ private enum SearchFolders { }
 				if (!TryReadManifest(GetParentDir(mod.modFile.path), out var manifest))
 					continue;
 
-				WorkshopHelper.QueryHelper.GetDependenciesRecursive(manifest.workshopEntryId, ref dependencies);
+				try {
+					WorkshopHelper.QueryHelper.GetDependenciesRecursive(manifest.workshopEntryId, ref dependencies);
+				}
+				catch (Exception e) {
+					Logging.tML.Warn($"Failed to fetch dependencies for local mod: {mod.DisplayName}. Check your internet connection!", e);
```

Comment:
Let's follow your advice to have this go to a UI indicating what happened; you are actually fairly close.

Use `Utils.ShowFancyErrorMessage("message", Interface.loadModsID)`.
We added this method a long time ago due to how often we need Error UI messages in the codebase.
Interface.loadModsID is the UI State when this gets called; resetting it will continue without this check (this time),

Update the message to mention that the 'dependencies' are 'missing dependencies', and that you can check the Steam Workshop to manually download them there.